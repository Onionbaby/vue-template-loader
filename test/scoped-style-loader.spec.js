const loader = require('../lib/scoped-style-loader')

describe('scoped style loader', () => {
  it('should add scoped id into the last node for each selector', done => {
    test(
      '1',
      'h1, h2 .foo {} h3 {}'
    ).then(done)
  })

  it('should not add scoped id into pseudo element/class', done => {
    test(
      '1',
      'p::before {} .test:first-child {}'
    ).then(done)
  })

  it('should add scoped id into the selectors in at-rules', done => {
    test(
      '1',
      '@media screen { p {} }'
    ).then(done)
  })

  it('should add scope attribute the selector before >>> combinator', done => {
    test(
      '1',
      '.foo .bar >>> .baz {}'
    ).then(done)
  })

  it('should add scope id to keyframes', done => {
    const id = 'abc'
    const input = [
      '.foo { animation: test 1s; }',
      '.bar { animation-name: test; animation-duration: 1s; }',
      '@keyframes test {',
      '  0% { opacity: 0; }',
      '  100% { opacity: 1; }',
      '}',
      '@-webkit-keyframes test {',
      '  from { opacity: 0; }',
      '  to { opacity: 1; }',
      '}'
    ].join('\n')
    test(id, input).then(done)
  })
})

function test (id, input) {
  return new Promise(resolve => {
    loader.call(
      {
        query: { id },
        resourcePath: 'test.css',
        async: () => (err, css) => {
          expect(css).toMatchSnapshot()
          resolve()
        },
        sourceMap: true
      },
      input
    )
  })
}

