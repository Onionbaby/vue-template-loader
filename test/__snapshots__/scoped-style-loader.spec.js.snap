// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scoped style loader should add scope attribute the selector before >>> combinator 1`] = `
"
.foo .bar[data-v-1] .baz {
}"
`;

exports[`scoped style loader should add scope id to keyframes 1`] = `
"
.foo[data-v-abc] { animation: test-data-v-abc 1s;
}
.bar[data-v-abc] { animation-name: test-data-v-abc; animation-duration: 1s;
}
@keyframes test-data-v-abc {
0% { opacity: 0;
}
100% { opacity: 1;
}
}
@-webkit-keyframes test-data-v-abc {
from { opacity: 0;
}
to { opacity: 1;
}
}"
`;

exports[`scoped style loader should add scoped id into the last node for each selector 1`] = `
"
h1[data-v-1], h2 .foo[data-v-1] {
}
h3[data-v-1] {
}"
`;

exports[`scoped style loader should add scoped id into the selectors in at-rules 1`] = `
"
@media screen {
p[data-v-1] {
}
}"
`;

exports[`scoped style loader should not add scoped id into pseudo element/class 1`] = `
"
p[data-v-1]::before {
}
.test[data-v-1]:first-child {
}"
`;
